app/classes/Transvision/Strings.php

```diff
@@ -64,15 +64,36 @@ public static function endsWith($haystack, $needles)
     }
 
     /**
-     * Check if $needle is in $haystack
+     * Check if $needles are in $haystack
      *
-     * @param  string  $haystack String to analyse
-     * @param  string  $needle   The string to look for
-     * @return boolean True if the $haystack string contains $needle
+     * @param string  $haystack  String to analyze
+     * @param mixed   $needles   The string (or array of strings) to look for
+     * @param boolean $match_all True if we need to match all $needles, false
+     *                           if it's enough to match one. Default: false
+     *
+     * @return boolean True if the $haystack string contains any/all $needles
      */
-    public static function inString($haystack, $needle)
+    public static function inString($haystack, $needles, $match_all = false)
     {
-        return mb_strpos($haystack, $needle, $offset = 0, 'UTF-8') !== false ? true : false;
+        $total_needles = count($needles);
+        $missing_needles = 0;
+
+        foreach ((array) $needles as $needle) {
+            if (mb_strpos($haystack, $needle, $offset = 0, 'UTF-8') !== false) {
+                // If I need to match any needle, I can stop at the first found
+                if (! $match_all) {
+                    return true;
+                }
+            } else {
+                $missing_needles++;
+            }
+        }
+
+        if ($match_all && $missing_needles == 0) {
+            return true;
+        }
+
+        return false;
```

Comment:
That seems a bit complicated, also $total_needles isn't used. I would rather write that:
```php
    public static function inString($haystack, $needles, $match_all = false)
    {
        $match = 0;
        foreach ((array) $needles as $needle) {
            if (mb_strpos($haystack, $needle, $offset = 0, 'UTF-8') !== false) {
                // If I need to match any needle, I can stop at the first found
                if (! $match_all) {
                    return true;
                }
                $match++;
            }
        }

        return $match == count($needles);
    }
```